src/Sylius/Bundle/TaxonomyBundle/Form/Type/TaxonChoiceType.php

```diff
@@ -65,27 +108,25 @@ public function configureOptions(OptionsResolver $resolver)
     {
         $repository = $this->taxonRepository;
         $choiceList = function (Options $options) use ($repository) {
-            $taxons = $repository->getNonRootTaxons();
-
-            if (null !== $options['taxonomy']) {
-                $taxons = $repository->getTaxonsAsList($options['taxonomy']);
-            }
+            $taxons = $repository->getChildren($options['root'], $options['direct']);
```

Comment:
When I look at Gedmo implementation of this method I feel like we should create our own implementation... 

Simple implementation does the job:
```php
    public function findChildren(TaxonInterface $taxon)
    {
        $queryBuilder = $this->createQueryBuilder('o');
        $queryBuilder
            ->andWhere(':left < o.left AND o.right < :right')
            ->setParameter('left', $taxon->getLeft())
            ->setParameter('right', $taxon->getRight())
        ;

        return $queryBuilder->getQuery()->getResult();
    }
```